src/happy_eyeballs.ml

```diff
@@ -103,23 +110,40 @@ let expand_list_split ips ports =
   | _ -> failwith "ips or ports are empty"
 
 let tick t now host id conn =
-  match
-    match conn.state with
-    | Resolving when Int64.sub now conn.created > t.resolve_timeout ->
-      Error () (* TODO retry resolution *)
-    | Waiting_for_aaaa (started, ips) when Int64.sub now started > t.aaaa_timeout ->
-      let ips = List.map (fun ip -> Ipaddr.V4 ip) (Ipaddr.V4.Set.elements ips) in
-      let dst, dsts = expand_list_split ips conn.ports in
-      Ok (Connecting (now, dst, dsts), [ Connect (host, id, dst) ])
-    | Connecting (started, _dst, dsts) when Int64.sub now started > t.connect_timeout->
+  match conn.state with
+  | Resolving when Int64.sub now conn.created > t.resolve_timeout ->
+    if conn.resolve_left = 0 then
+      Error ()
+    else
+      Ok ({ conn with created = now ; resolve_left = conn.resolve_left - 1 },
+          [ Resolve_a host ; Resolve_aaaa host ])
```

Comment:
This seems to imply `resolve_retries` of 3 will mean we attempt to resolve up to 4 times since we start out in `Resolving` state with `resolve_left` of `resolve_retries`. It makes resolve_**re**tries very literal, and perhaps we should emphasize this in the documentation.

Reply:
I also fixed this